scylla/src/transport/iterator.rs

```diff
@@ -540,6 +540,27 @@ where
                     self.metrics.inc_failed_paged_queries();
                     return Err(err);
                 }
+                Ok(NonErrorQueryResponse {
+                    response: NonErrorResponse::Result(_),
+                    ..
+                }) => {
+                    // We have most probably sent a modification statement (e.g. INSERT or UPDATE),
+                    // so let's return an empty iterator as suggested in #631.
+
+                    // We must attempt to send something because the iterator expects it.
+                    let (proof, _) = self
+                        .sender
+                        .send(Ok(ReceivedPage {
+                            rows: Rows {
+                                metadata: Default::default(),
+                                rows_count: 0,
+                                rows: Vec::new(),
+                            },
+                            tracing_id: None,
```

Comment:
`tracing_id` is available in the `NonErrorQueryResponse`, so let's use it.

Reply:
How do I test that? (as you mentioned in the general comment)

Reply:
Run `query_iter` with tracing enabled (`Query::set_tracing`) and then verify that the list of tracing ids reported by the iterator is not empty.